docs/source/conf.py

```diff
@@ -96,3 +96,4 @@
 html_permalinks = False
 html_show_sphinx = False
 todo_include_todos = True
+mermaid_output_format = "png"
```

Comment:
How/why does this help?

Reply:
mermaid handles things differently looking at output format changing the code to embed to html etc. with image png it is not changing anything with theme changes and you can easily see things inside basically, with svg it is using different thing and you need to customize everything.

Reply:
should be much better right @istankovic ?

Reply:
hard to understand i can see it normally. is there a chance it is using css classes or something in different browsers? which browser are you using? i am using chrome and it was ok.

Reply:
I'm using Firefox. With Chromium I get it the way it's supposed to be (by the way, much nicer!):

![image](https://user-images.githubusercontent.com/1312091/233582549-c49d7970-ef75-4ddc-9ea3-0c5236b5d48d.png)

Reply:
yeah should be fine now. we had a look with @GabrielBuragev and found a much better css to handle this.

Reply:
> yeah should be fine now. we had a look with @GabrielBuragev and found a much better css to handle this.

Hmmm, I still have the same issue with my Firefox on ad56603f90ff8a816891b0059583fe53f6111714.
I did a `make clean` before `make docs`.

Reply:
i think you can check this again

Reply:
This is what I get with af5a564db39aa71a5a9bfde695816c43389da107:
![image](https://user-images.githubusercontent.com/1312091/233723454-f17a513a-73f6-4d10-a1fc-6c84de640fe3.png)

Reply:
Compared to the Chromium's rendering, Firefox is more blurry, almost looks like bold.
I have no idea what the problem might be, but let's just blame Firefox or my local setup for it.
(It would be nice if the dark theme worked nicely on all major browsers, though. :disappointed: )

Reply:
@bilbeyt is there something we could do about the ugly text rendering in Firefox?

Reply:
@bilbeyt I also see the texts blurred as on the screenshot Ivan provided. 
The problem here is that the css holds these values for the texts:
`fill: #333;`

This in fact fills the text with black color. 
You might need to also overwrite this fill styling by expanding the css for the message elements like so:

```
body:not([data-theme="light"]) .mermaid .messageText {
        stroke: white !important;
        fill: white !important;
    }
```

Reply:
After adding the above, the text should look like this:

![Screenshot 2023-04-24 at 10 26 42](https://user-images.githubusercontent.com/18466333/233941294-ef5b373b-2c20-4a0c-8fcd-c7a6242f3cd2.png)

Reply:
much better thanks @GabrielBuragev will add it also

Reply:
LGTM ✅